motoko/canister_logs/Makefile

```diff
@@ -20,9 +20,35 @@ upgrade: build
 .PHONY: test
 .SILENT: test
 test: install
-	dfx canister call CanisterLogs print 'hi'
+	# Test print via update call.
+	dfx canister call CanisterLogs print 'print via update'
 	dfx canister logs CanisterLogs \
-		| grep 'hi' && echo 'PASS'
+		| grep 'print via update' && echo 'PASS'
+
+	# Test print via replicated query call.
+	dfx canister call --update CanisterLogs print_query 'print via replicated query'
+	dfx canister logs CanisterLogs \
+		| grep 'print via replicated query' && echo 'PASS'
+
+	# Test print via non-replicated query call should NOT record the message.
+	dfx canister call --query CanisterLogs print_query 'print via non-replicated query'
+	! dfx canister logs CanisterLogs \
+		| grep 'print via non-replicated query' && echo 'PASS'
+
+	# Test trap, ignore failed call output.
+	-dfx canister call CanisterLogs trap 'trap via update'
```

Comment:
why `-dfx`, also below?

Reply:
because trapped call fails which makes the `make test` to fail as well. but in this case this failure is expected, we need this call to trap and then see that the trap message appears in logs.

Reply:
But I cannot just run
```
$ -dfx
-dfx: command not found
```
so how does that `-dfx` work?

Reply:
> To ignore errors in a recipe line, write a ‘-’ at the beginning of the line’s text (after the initial tab). The ‘-’ is discarded before the line is passed to the shell for execution.

it's a GNU make synthax, see [here](https://www.gnu.org/software/make/manual/make.html#Errors)

Reply:
@mraszyk sorry, I merged the PR before addressing this comment. 

here's another cleanup PR, please take a look.

https://github.com/dfinity/examples/pull/929